library/std/src/sys/unix/rand.rs

```diff
@@ -26,10 +25,10 @@ mod imp {
     use crate::fs::File;
     use crate::io::Read;
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "freebsd"))]
     use crate::sys::weak::syscall;
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "freebsd"))]
```

Comment:
`GRND_INSECURE` seems to be meaningless on FreeBSD. (It's non-existent on FreeBSD 12, and equivalent to `GRND_NONBLOCK` on FreeBSD 13+.) So re-using this same logic with `GRND_INSECURE_AVAILABLE` for Linux doesn't seem very useful.

Reply:
good point